### ISSUE TITLE
Shared folders not working on latest VMWare Fusion

### ISSUE THREAD

**Issue by khromov** [wplib/wplib-box#98]
There was recently a patch for VMWare Fusion provider to fix some long-standing issues with shared folders, backstory here: 
https://github.com/mitchellh/vagrant/issues/6775

After updating to the latest version of VMWare Fusion provider (4.0.9) the box doesn't boot with the error:

```
==> default: Waiting for HGFS to become available...
==> default: Enabling and configuring shared folders...
An error occurred while executing `vmrun`, a utility for controlling
VMware machines. The command and output are below:

Command: ["enableSharedFolders", "/Users/stakhr/wplib-box/.vagrant/machines/default/vmware_fusion/41423efe-4080-449e-9732-93a790df5a92/wplib-box.vmx", {:notify=>[:stdout, :stderr]}]

Stdout: Error: There was an error mounting the Shared Folders file system inside the guest operating system

Stderr: 

```

**Comment by mikeschinkel** [comment]
@khromov So we need to update our box?  

New version coming first of this week, hopefully Monday.

**Comment by clubdeuce** [comment]
@khromov I am unable to replicate this error. I am using VMWare Fusion Version 8.1.1 (3771013) and Vagrant-VMWare Plugin 4.0.9.

Are you still experiencing this issue?

**Comment by ksquaredkey** [comment]
I am having the exact same issue.  I ssh'ed into my Vagrant and repeated the command by hand and I get an exit status of 255.  VMware Fusion Pro 8.1.1, fusion plugin 4.0.9, guest is a CentOS 6.6 box I build in December with Fusion Pro 8.0.x

>==> default: Enabling and configuring shared folders...
>   default: -- /Users/key/Dev/Stash/Server-stuff: /repos
>The following SSH command responded with a non-zero exit status.
>Vagrant assumes that this means the command failed!
>
>vmhgfs-fuse -o uid=`id -u vagrant`,gid=500,allow_other '.host:/-repos' /repos
>
>Stdout from the command:
>
>
>Stderr from the command:
>
>

**Comment by clubdeuce** [comment]
@ksquaredkey What version of Vagrant are you using?

**Comment by ksquaredkey** [comment]
$ vagrant --version
Vagrant 1.8.1

**Comment by ksquaredkey** [comment]
Ah, I see 1.8.4 is out - I will update and retry.

**Comment by ksquaredkey** [comment]
Same result.

**Comment by ksquaredkey** [comment]
I got rid of my relative paths, it still blows up on:

`  default: -- /Users/key/Dev/Stash/Server-stuff/cortext-dev-tools/vm: /vagrant
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

vmhgfs-fuse -o uid=`id -u vagrant`,gid=500,allow_other '.host:/-vagrant' /vagrant

Stdout from the command:



Stderr from the command:

`

**Comment by ksquaredkey** [comment]
Sorry, just noticed this is not the vagrant issues - got to this issue from a link from there and failed to double-check where I landed.  I'll take my issue to Vagrant.

**Comment by clubdeuce** [comment]
Since  there has been no response from @khromov and I cannot replicate, I am closing this issue.

**Comment by Patryk1098** [comment]
So I've been running into this same issue while building a vagrant box. I was able to resolve this issue by purging the open-vm-tools package within the guest OS (ubuntu server 16.04), and installing the proprietary vmware-tools that VMWare Fusion provides.

Following all the defaults of the install script (aside from one), I was able to resolve this issue. Basically; it's more likely a bug in the guest OS.

Specific things I did:
`apt-get purge open-vm-tools`
`# mount cdrom with vmware-tools`
`# copy all files on cd to /root`
`tar zxf VMware....tar.gz`
`cd vmware-tools-distrib`
`./vmware-install.pl`
`# follow all defaults -making sure to install vmhgfs, BUT enable dynamic kernel module loading`
`modprobe vmhgfs`
`lsmod | grep hgfs`

Try mounting shared folders again --should work.

If this doesn't resolve your problem, or it's unrelated, I apologize for the spam. :)
